src/Symfony/Bridge/Doctrine/Form/ChoiceList/ORMQueryBuilderLoader.php

```diff
@@ -55,6 +55,26 @@ public function getEntities()
      */
     public function getEntitiesByIds($identifier, array $values)
     {
+        if (null !== $this->queryBuilder->getMaxResults()) {
+            $metadata = $this->queryBuilder->getEntityManager()->getClassMetadata(current($this->queryBuilder->getRootEntities()));
```

Comment:
The only other way I see to fix this is to throw an exception here and to catch it in the `DoctrineChoiceLoader` to let it fall back on loading the list. But I would prefer to have it part of the `EntityLoaderInterface` contract rather than leaking too much in there.